### PR TITLE
fix: decrypt remote parameter for SecureString type

### LINKED ISSUE #3697: ESO keeps pushing the same parameter for every sync when it's type is `SecureString`
**Describe the bug**
When pushing an encrypted parameter (through KMS) to `AWS Parameter Store`, ESO currently creates different versions from the same parameter on each sync interval. This happens because ESO compares a remote encrypted parameter value with a non-encrypted one, resulting in a new push.

**To Reproduce**
1. Create a `PushSecret` to push a parameter with `SecureString` parameter type.
3. Watch the new version of the same secret to be created in every ESO iteration

**To Fix**
Guarantee that [ssm.GetParameterInput](https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/aws/parameterstore/parameterstore.go#L195) is setting `WithDecryption`. Example:

```
secretValue := ssm.GetParameterInput{
  Name:           &secretName,
  WithDecryption: aws.Bool(true),
}
```

**Expected behavior**
I expect ESO to update the remote secret only when the secret value is different from the remote one. 

**Screenshots**
bug_evidence

**Additional context**
This bug was introduced after this [PR](https://github.com/external-secrets/external-secrets/pull/3576) added support to push parameters with `SecureString` values.

### PR DESCRIPTION
**Pull request by vsantos**
## Problem Statement

This PR fixes a `PushSecret` being updated remotely in every ESO cycle in case of a parameter with type `SecureString`.

## Related Issue

Fixes #3697

## Proposed Changes

Ensure that `WithDecryption` is enabled when fetching a remote parameter value for type `SecureString`.

## Checklist

- [X] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [X] All commits are signed with `git commit --signoff`
- [X] My changes have reasonable test coverage
- [X] All tests pass with `make test`
- [X] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

File: pkg/provider/aws/parameterstore/parameterstore.go [hunk 1]
```diff
@@ -195,7 +195,8 @@ func (pm *ParameterStore) PushSecret(ctx context.Context, secret *corev1.Secret,
 	}
 
 	secretValue := ssm.GetParameterInput{
-		Name: &secretName,
+		Name:           &secretName,
+		WithDecryption: aws.Bool(true),
```
[the remark this answers is not in the document]
Comment (vsantos) [reply pkg/provider/aws/parameterstore/parameterstore.go:199]: Hi @Skarlso, thanks for this input. `WithDecryption` will only be used for `SecureString` specifically, which means it doesn't affect any other parameter type such as `String`, preserving the same behavior as before:

>  
    // Return decrypted values for secure string parameters. This flag is ignored
    // for String and StringList parameter types.
    WithDecryption *bool `type:"boolean"`

There are three types of `types`: `SecureString`, `String`, and `StringList`. Without `WithDecryption`, when fetching a `SecureString` value, the value will be a hardcoded `sensitive` string.

[the remark this answers is not in the document]
Comment (Skarlso) [reply pkg/provider/aws/parameterstore/parameterstore.go:199]: Ah okay good. Thanks for the clarification.

**Review by Skarlso — APPROVED** [review]

**Pull request merged**